ols/app/models/config.py

```diff
@@ -120,6 +120,44 @@ def validate_yaml(self, disable_tls: bool = False) -> None:
             checks.file_check(self.tls_key_path, "OLS server certificate private key")
 
 
+class ProxyConfig(BaseModel):
+    """HTTPS Proxy configuration."""
+
+    proxy_url: Optional[AnyHttpUrl] = None
+    proxy_ca_cert_path: Optional[FilePath] = None
+
+    def __init__(self, data: Optional[dict] = None) -> None:
+        """Initialize configuration and perform basic validation."""
+        super().__init__()
+        if not data:
+            return
+        self.proxy_url = data.get("proxy_url")  # type: ignore
+        self.proxy_ca_cert_path = data.get("proxy_ca_cert_path", None)
+
+    def validate_yaml(self) -> None:
+        """Validate proxy config."""
+        if self.proxy_url is None and self.proxy_ca_cert_path:
+            raise checks.InvalidConfigurationError("Proxy URL is missing")
+        if self.proxy_url and not checks.is_valid_http_url(self.proxy_url):
+            raise checks.InvalidConfigurationError(
+                "Proxy URL is invalid, only http:// and https:// URLs are supported"
+            )
+        if self.proxy_ca_cert_path is not None:
+            checks.file_check(self.proxy_ca_cert_path, "Proxy CA certificate")
+
+    @classmethod
+    def from_dict(cls, data: Optional[dict] = None) -> Self | None:
+        """Create a ProxyConfig instance from a dictionary."""
+        os_proxy_url = os.environ.get("https_proxy")
```

Comment:
I see there is some exception to have the env `https_proxy` in lowercase. The lowercase have precedense over uppercase. Would it work if we define the env name in standard uppercase format?

Reply:
this implementation works in lower case only, will add a upper case.